test/unit/publicapi.test.js

```diff
@@ -156,12 +156,18 @@ suite('Public API', function() {
       mq.latex('a x^2 + b x + c = 0');
       assert.equal(mq.__controller.cursor[L].ctrlSeq, '0');
       assert.equal(mq.__controller.cursor[R], 0);
+      assert.ok(mq.cursorAtRightEnd());
+      assert.ok(!mq.cursorAtLeftEnd());
       mq.moveToLeftEnd();
       assert.equal(mq.__controller.cursor[L], 0);
       assert.equal(mq.__controller.cursor[R].ctrlSeq, 'a');
+      assert.ok(mq.cursorAtLeftEnd());
+      assert.ok(!mq.cursorAtRightEnd());
       mq.moveToRightEnd();
       assert.equal(mq.__controller.cursor[L].ctrlSeq, '0');
       assert.equal(mq.__controller.cursor[R], 0);
+      assert.ok(mq.cursorAtRightEnd());
+      assert.ok(!mq.cursorAtLeftEnd());
```

Comment:
**nit**: I think (at least when I read it) I'd expect the Left and Right assertions to always be in the same order for these pairs of assertions.  I had to take a second look to notice that the middle pair were reversed and that's why they could be `.ok(x)` and `.ok(!x)`.    This also matches the `[L]` `[R]` ordering that precedes each pair of assertions. 

I know this is extremely nitpicky, but these types of things add just tiny bits of friction to a developers life.  I'm not sure everyone thinks the same though so it's your call. :) 

```suggestion
      assert.ok(!mq.cursorAtLeftEnd());
      assert.ok(mq.cursorAtRightEnd());
      mq.moveToLeftEnd();
      assert.equal(mq.__controller.cursor[L], 0);
      assert.equal(mq.__controller.cursor[R].ctrlSeq, 'a');
      assert.ok(mq.cursorAtLeftEnd());
      assert.ok(!mq.cursorAtRightEnd());
      mq.moveToRightEnd();
      assert.equal(mq.__controller.cursor[L].ctrlSeq, '0');
      assert.equal(mq.__controller.cursor[R], 0);
      assert.ok(!mq.cursorAtLeftEnd());
      assert.ok(mq.cursorAtRightEnd());
```

Reply:
Nah, that makes sense! I was trying to find a more readable way to assert these, but I am not totally familiar with Mocha's TDD API, so I went with what I could find.